AntennaTracker/control_auto.cpp

```diff
@@ -58,3 +70,57 @@ void Tracker::calc_body_frame_target(float pitch, float yaw, float& bf_pitch, fl
     bf_pitch   = ahrs.cos_roll()     * pitch    + ahrs.sin_roll()  * ahrs.cos_pitch() * yaw;
     bf_yaw     = -ahrs.sin_roll()    * pitch    + ahrs.cos_pitch() * ahrs.cos_roll()  * yaw;
 }
+bool Tracker::convert_bf_to_ef(float pitch, float yaw, float& ef_pitch, float& ef_yaw)
+{
+    // avoid divide by zero
+    if (is_zero(ahrs.cos_pitch())) {
+        return false;
+    }
+    // convert earth frame angle or rates to body frame
+    ef_pitch =  ahrs.cos_roll()                     * pitch -  ahrs.sin_roll()                     * yaw; //pitch
+    ef_yaw =   (ahrs.sin_roll() / ahrs.cos_pitch()) * pitch + (ahrs.cos_roll() / ahrs.cos_pitch()) * yaw; //yaw
+    return true;
+}
+void Tracker::set_servo_direction(bool& direction_reversed)
+{
+    // calculating distances from current pitch/yaw to lower and upper limits in centi-degrees
+    float yaw_angle_limit_lower =   100 * (channel_yaw.get_servo_out()-(-9000))  / (9000-(-9000)) * g.yaw_range;
```

Comment:
I think these formulas are not quite right.  I think it should be more like this:
yaw_angle_limit_lower = (-g.yaw_range * 100.0f / 2.0f) - channel_yaw.get_servo_out()
yaw_angle_limit_lower = (g.yaw_range * 100.0f / 2.0f) - channel_yaw.get_servo_out()